# Homework 5:

## Notebook by Creston Wilson 11/10/2017

## Created for CIS 399, Probability and Statistics for Computer Science

### References/Acknowledgement: Professor Lowd's code samples

### Part A:

#### Question 3.24
##### Speeding on the I-5, Part II

Exercise 3.12 states that the distribution of speeds of cars traveling on the Interstate 5 Freeway (I-5) in California is nearly normal with a mean of 72.6 miles/hour and a standard deviation of 4.78 miles/hour. The speed limit on this stretch of the I-5 is 70 miles/hour.

    (a) A highway patrol officer is hidden on the side of the freeway. What is the probability that 5 cars pass and none are speeding? Assume that the speeds of the cars are independent of each other.

.7054 is the probability that a car is driving over 70 mile/hour. So then the probability that none of the five cars pass are speeding is represented by:  (1 - .7054)^5 = .0022  or .22% probability that five cars in a row aren't speeding.

    (b) On average, how many cars would the highway patrol officer expect to watch until the first car that is speeding? What is the standard deviation of the number of cars he would expect to watch?
    
We are looking for a mean of a Geometric Distribution. So the average number of cars expected to be seen before one is speeding is 1.4 cars.

#### Question 3.26
##### Chicken pox, Part 1

The National Vaccine Information Center estimates that 90% of Americans have had chickenpox by the time they reach adulthood.

    (a) Is the use of the binomial distribution appropriate for calculating the probability that exactly 97 out of 100 randomly sampled American adults had chickenpox during childhood.

Yes, the binomial distribution describes the probability of having exactly k successes in n independent Bernoulli trials with probability of a success p.

    (b) Calculate the probability that exactly 97 out of 100 randomly sampled American adults had chickenpox during childhood.

The probability that 97 out of 100 adults exactly had chickenpox during childhood is .59%.

    (c) What is the probability that exactly 3 out of a new sample of 100 American adults have not had chickenpox in their childhood?
 
The probability of this question is the same as (b), .59%.
 
    (d) What is the probability that at least 1 out of 10 randomly sampled American adults have had chickenpox?
  
This is equivalent to the expression: P(all outcomes)-P(that no adults have had chickenpox).  The probability that at least 1 out of 10 adults have had chickenpox is 100%.
  
    (e) What is the probability that at most 3 out of 10 randomly sampled American adults have not had chickenpox?

This is equivalent to the expression: P(0 adults no chickenpox) + P(1 adult no chickenpox) + P(2 adults no chickenpox) + P(3 adults no chickenpox). This equals   


#### Question 3.30
##### Survey response rate

Pew Research reported in 2012 that the typical response rate to their surveys is only 9%. If for a particular survey 15,000 households are contacted, what is the probability that at least 1,500 will agree to respond?

This is equivalent to the expression: 1 - P(1500 agreeing to respond).
Which is equal to 99.99%.


#### Question 4.2
##### Identify the parameter, Part II

For each of the following situations, state whether the parameter of interest is a mean or a proportion.

    (a) A poll shows that 64% of Americans personally worry a great deal about federal spending and the budget deficit.

Proportion

    (b) A survey reports that local TV news has shown a 17% increase in revenue between 2009 and 2011 while newspaper revenues decreased by 6.4% during this time period.

Mean

    (c) In a survey, high school and college students are asked whether or not they use geolocation services on their smart phones.

Proportion

    (d) In a survey, smart phone users are asked whether or not they use a web-based taxi service.

Proportion

    (e) In a survey, smart phone users are asked how many times they used a web-based taxi service over the last year.
    
Mean

#### Question 4.8
##### Twitter users and news, Part I

A poll conducted in 2013 found that 52% of U.S. adult Twitter users get at least some news on Twitter. The standard error for this estimate was 2.4%, and a normal distribution may be used to model the sample proportion. Construct a 99% confidence interval for the fraction of U.S. adult Twitter users who get some news on Twitter, and interpret the confidence interval in context.

52% is our mean and so the 99% confidence interval for this data is (45.808%, 58.192%).


#### Question 4.14
##### Thanksgiving spending, Part I

The 2009 holiday retail season, which kicked off on November 27, 2009 (the day after Thanksgiving), had been marked by somewhat lower self-reported consumer spending than was seen during the comparable period in 2008. To get an estimate of consumer spending, 436 randomly sampled American adults were surveyed. Daily consumer spending for the six-day period after Thanksgiving, spanning the Black Friday weekend and Cyber Monday, averaged 84.71. A 95% confidence interval based on this sample is (80.31, 89.11). Determine whether the following statements are true or false, and explain your reasoning.

    (a) We are 95% confident that the average spending of these 436 American adults is between 80.31 and 89.11.

True, this is the information that the 95% confidence interval provides.

    (b) This confidence interval is not valid since the distribution of spending in the sample is right skewed.

False, the sample is strongly right skewed but this is made up for by a large sample of 436. The book reccomends using at least 100 observations as a sample size when strong skewing occurs.

    (c) 95% of random samples have a sample mean between 80.31 and 89.11.

True, this means that 5% of the time we take a random sample our sample mean will not fall into the confidence interval.

    (d) We are 95% confident that the average spending of all American adults is between 80.31 and 89.11.

True, this is why we take the confidence interval is to determine how likely we are to capture the population mean when taking a random sample.

    (e) A 90% confidence interval would be narrower than the 95% confidence interval since we don’t need to be as sure about our estimate.

False, it would be wider because we would be less accurate as to where the true mean is located.

    (f) In order to decrease the margin of error of a 95% confidence interval to a third of what it is now, we would need to use a sample 3 times larger.
 
False the number of samples is under a squareroot so in order to get a third we need to use a sample 9 times larger.
 
    (g) The margin of error is 4.4.
    
True, this can be found by subtracting either end of the interval with the sample mean and taking the absolute value of that number.

#### Question 4.16
##### Age at first marriage, Part I

The National Survey of Family Growth conducted by the Centers for Disease Control gathers information on family life, marriage and divorce, pregnancy, infertility, use of contraception, and men’s and women’s health. One of the variables collected on this survey is the age at first marriage. The histogram below shows the distribution of ages at first marriage of 5,534 randomly sampled women between 2006 and 2010. The average age at first marriage among these women is 23.44 with a standard deviation of 4.72.

    Estimate the average age at first marriage of women using a 95% confidence interval, and interpret this interval in context. Discuss any relevant assumptions.

In order to calculate the 95% CI we assume that the observations are independent. The sample is large enough that the right skew in the data shouldn't affect the results to bad. The 95% confidence interval for a sample mean of 23.44 and a standard deviation of 4.72 is (23.31, 23.57).
We are 95% confident that the average age at first marriage among women is between 

#### Part B:

In [297]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

%matplotlib inline
responses = pd.read_csv("multipleChoiceResponses.csv", encoding="iso-8859-1", low_memory = False)

#### Question 1
The average age over all survey respondents is (approximately) 32.4. On average, are respondents who are “Employed by government” older or younger than 32.4?

In [298]:
govemploy = responses[(responses.CurrentEmployerType=='Employed by government')]
govemploy.describe()

Age  LearningCategorySelftTaught  LearningCategoryOnlineCourses  \
count  530.000000                   510.000000                     509.000000   
mean    36.300000                    32.919608                      26.868369   
std     10.353424                    27.535571                      28.109046   
min      0.000000                     0.000000                       0.000000   
25%     29.000000                    10.000000                       4.000000   
50%     35.000000                    25.000000                      20.000000   
75%     42.000000                    50.000000                      40.000000   
max     72.000000                   100.000000                     100.000000   

       LearningCategoryWork  LearningCategoryUniversity  \
count            508.000000                  508.000000   
mean              15.314697                   18.309055   
std               18.960167                   24.538682   
min                0.000000                    0.000000   
25%                0.000000                    0.000000   
50%               10.000000                    5.000000   
75%               25.000000                   30.000000   
max               90.000000                  100.000000   

       LearningCategoryKaggle  LearningCategoryOther  TimeGatheringData  \
count              510.000000             510.000000         334.000000   
mean                 5.805882               0.966929          32.895210   
std                 12.441911               5.867930          21.708783   
min                  0.000000               0.000000           0.000000   
25%                  0.000000               0.000000          20.000000   
50%                  0.000000               0.000000          30.000000   
75%                 10.000000               0.000000          50.000000   
max                100.000000              80.000000         100.000000   

       TimeModelBuilding  TimeProduction  TimeVisualizing  \
count         333.000000      334.000000       334.000000   
mean           20.303303        9.308383        15.428144   
std            17.474109       10.868203        13.497834   
min             0.000000        0.000000         0.000000   
25%            10.000000        0.000000         8.500000   
50%            20.000000        5.000000        10.000000   
75%            30.000000       14.500000        20.000000   
max           100.000000       80.000000       100.000000   

       TimeFindingInsights  TimeOtherSelect  
count           334.000000       333.000000  
mean             15.377246         4.759760  
std              14.347727        17.055966  
min               0.000000         0.000000  
25%               5.000000         0.000000  
50%              10.000000         0.000000  
75%              20.000000         0.000000  
max             100.000000       100.000000

Looking at the mean of the Age column for all the government employed respondents shows that the average government employee age is over 32.4.
In fact it is 36.3.

#### Question 2
The average age of survey respondents from the United States is (approximately) 35.2. On average, are respondents in Canada older or younger than 35.2?

In [299]:
canada = responses[responses.Country == 'Canada']
canada.describe()

Age  LearningCategorySelftTaught  LearningCategoryOnlineCourses  \
count  431.000000                   341.000000                     342.000000   
mean    35.772622                    35.903226                      26.251462   
std     11.912053                    27.170367                      26.275686   
min     13.000000                     0.000000                       0.000000   
25%     27.000000                    19.000000                       5.000000   
50%     34.000000                    30.000000                      20.000000   
75%     43.500000                    50.000000                      40.000000   
max     82.000000                   100.000000                     100.000000   

       LearningCategoryWork  LearningCategoryUniversity  \
count            342.000000                  341.000000   
mean              13.979532                   18.870968   
std               17.030838                   24.975654   
min                0.000000                    0.000000   
25%                0.000000                    0.000000   
50%               10.000000                   10.000000   
75%               25.000000                   30.000000   
max               80.000000                  100.000000   

       LearningCategoryKaggle  LearningCategoryOther  TimeGatheringData  \
count              341.000000             341.000000         210.000000   
mean                 4.363636               0.806452          35.409524   
std                  8.644772               6.394673          22.720560   
min                  0.000000               0.000000           0.000000   
25%                  0.000000               0.000000          20.000000   
50%                  0.000000               0.000000          30.000000   
75%                  5.000000               0.000000          50.000000   
max                 60.000000              80.000000         100.000000   

       TimeModelBuilding  TimeProduction  TimeVisualizing  \
count         210.000000      210.000000       210.000000   
mean           21.857143       10.447619        13.309524   
std            18.163757       13.048662        11.399001   
min             0.000000        0.000000         0.000000   
25%            10.000000        0.000000         5.000000   
50%            20.000000       10.000000        10.000000   
75%            30.000000       15.000000        20.000000   
max           100.000000       80.000000       100.000000   

       TimeFindingInsights  TimeOtherSelect  
count           210.000000       210.000000  
mean             12.428571         4.166667  
std              12.302298        16.653400  
min               0.000000         0.000000  
25%               5.000000         0.000000  
50%              10.000000         0.000000  
75%              20.000000         0.000000  
max              70.000000       100.000000

By looking at the count of all the Canadian respondents we can see that the mean for the Age column is just about 35.2. The average age for Canadian respondents is 35.772622.

#### Question 3
8.8% of respondents rate their JobSatisfaction as “10 - Highly Satisfied”. On average, are respondents who use Jupyter more or less likely to be highly satisfied (compared to 8.8%)?

In [300]:
jupyter_users = responses[(responses.WorkToolsFrequencyJupyter.notnull()) & (responses.JobSatisfaction.notnull())]
jupyter_users = jupyter_users[['JobSatisfaction', 'WorkToolsFrequencyJupyter']]
jupyter_users.describe()

JobSatisfaction WorkToolsFrequencyJupyter
count             2795                      2795
unique              11                         4
top                  8          Most of the time
freq               673                      1166

On average, respondents that use Jupyter are more likely to be highly satisfied because the size of samples being compared is smaller for Jupyter users than respondents that reported JobSatisfaction.

#### Question 4
On average, are respondents who use Excel more or less likely to be highly satisfied (compared to 8.8%)?

In [301]:
eu = responses[(responses.WorkToolsFrequencyExcel.notnull()) & (responses.JobSatisfaction.notnull())]
eu = eu[['JobSatisfaction', 'WorkToolsFrequencyExcel']]
eu.describe()

JobSatisfaction WorkToolsFrequencyExcel
count              911                     911
unique              11                       4
top                  8                   Often
freq               191                     323

On average, respondents that use Excel are more likely to be highly satisfied because the size of samples being compared is smaller for Excel users than respondents that reported JobSatisfaction.

#### Question 5
Come up with your own hypothesis and test it!